scripts/rust-is-available.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0
+#
+# Tests whether a suitable Rust toolchain is available.
+#
+# Pass `-v` for human output and more checks (as warnings).
+
+set -e
+
+min_tool_version=$(dirname $0)/min-tool-version.sh
+
+# Convert the version string x.y.z to a canonical 5 or 6-digit form.
+get_canonical_version()
+{
+	IFS=.
+	set -- $1
+	echo $((10000 * $1 + 100 * $2 + $3))
```

Comment:
This will break in a couple of years once rust 1.100.0 is released.

Reply:
Good catch!

Reply:
I will add a note that this is different from the other instances of this function in case someone copies it from here.

Reply:
It is late 2026 when that would happen, though, not a couple of years, no?

```python
>>> datetime.date.today() + datetime.timedelta(weeks=(100-58)*6)
datetime.date(2026, 11, 14)
```

Reply:
Also, since we are only using the value for comparison (unlike other version scripts), it would not really break unless Rust 2 gets released (and used!).